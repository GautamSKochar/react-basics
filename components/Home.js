import React from 'react';
import { connect } from 'react-redux';
import BlogPost from './BlogPost';
import { setName } from '../actions/userActions';

class Home extends React.Component {
    constructor(props) {
        super();
        this.state = {
            age: props.age,
            status: 0,
            homeLink: "MyHome",
            value: 'Type some *markdown* here!'
        };
        setTimeout(() => {
            this.setState({
                status: 1
            })
        }, 5000);
        this.handleChange = this.handleChange.bind(this);
        // this.age = props.age;
    }

    componentWillMount() {
        console.log('Component will mount');
    };
    componentDidMount() {
        console.log('Component did mount');
    };
    componentWillReceiveProps(nextProps) {
        console.log('Component will receive props', nextProps);
    };
    shouldComponentUpdate(nextProps, nextState) {
        console.log('Should Component update', nextProps, nextState);
        return true;
    }
    componentWillUpdate(nextProps, nextState) {
        console.log('Component will update', nextProps, nextState);
    }
    componentDidUpdate(prevProps, prevState) {
        console.log('Component Did Update', prevProps, prevState);
    }

    handleChange(e) {
        this.setState({ value: e.target.value });
    }

    onMakeOlder() {
        this.setState({
            age: parseInt(this.state.age) + 2
        })
        console.log(this.state.age);
    }
    onChangeLink() {
        this.props.changeLink(this.state.homeLink)
    }
    onHandleChange(event) {
        this.setState({
            homeLink: event.target.value
        })
    }
    render() {
        return (
            <div>
                <p>In a new Component!</p>
                <p>Your name is {this.props.user.name}, your age is {this.state.age}</p>
                <p>Status: {this.state.status}</p>
                <hr />
                <button onClick={this.onMakeOlder.bind(this)} className="btn btn-primary">Make me Older</button>&nbsp;
                <button onClick={() => this.props.setName("Yuvi")} className="btn btn-primary">Change Name</button>&nbsp;
                
                <hr />
                <div className="MarkdownEditor">
                    <h3>Input</h3>
                    <textarea
                        rows="4" cols="30" style={{ overflow: 'hidden' }}
                        onChange={this.handleChange}
                        defaultValue={this.state.value}
                    />
                    <h3>Output</h3>
                    <div className="content">
                        <p className="content">{this.state.value}</p>
                    </div>
                </div>
                <hr/><h4>Reading data from json file:</h4>
<BlogPost />
            </div>
        );
    }
}


const mapStateToProps = (state) => {
    return {
        user: state.user
    };
}

const mapDispatchToProps = (dispatch) => {
    return {
        setName: (name) =>{
            dispatch(setName(name))
        }
    };
}
export default connect(mapStateToProps,mapDispatchToProps)(Home)
//export default Home;