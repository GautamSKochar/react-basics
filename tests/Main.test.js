import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Home from '../components/Home';
// setup file
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

const raf = global.requestAnimationFrame = (cb) => {
    setTimeout(cb, 0)
}

configure({ adapter: new Adapter() });

test('Home Component should render', () => {
    const component = shallow(<Home />);
    const tree = toJson(component);
    console.log(tree);
    expect(component.contains(<p>In a new Component!</p>)).toBe(true);
    expect(tree).toMatchSnapshot();
    
    // it('renders without crashing', () => {
    //     const component = shallow(<Home />);
    //     expect (component).toHaveLength(1);
    //     // const div = document.createElement('div');
    //     // ReactDOM.render(<Home />, div);
    //   });
});    