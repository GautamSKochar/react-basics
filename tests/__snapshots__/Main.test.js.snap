// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component should render 1`] = `
<div>
  <p>
    In a new Component!
  </p>
  <p>
    Your name is 
    , your age is 
  </p>
  <p>
    Status: 
    0
  </p>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Make me Older
  </button>
  Â 
  <hr />
  <div
    className="MarkdownEditor"
  >
    <h3>
      Input
    </h3>
    <textarea
      cols="30"
      defaultValue="Type some *markdown* here!"
      onChange={[Function]}
      rows="4"
      style={
        Object {
          "overflow": "hidden",
        }
      }
    />
    <h3>
      Output
    </h3>
    <div
      className="content"
    >
      <p
        className="content"
      >
        Type some *markdown* here!
      </p>
    </div>
  </div>
  <BlogPost />
</div>
`;
